### ISSUE TITLE
SDK build error: lp_message_time_n?ow_with_state : <identifier> expected

### ISSUE THREAD

**Issue by bramley-turner-stride** [LP-Messaging/Android-Messaging-SDK#34]
I am trying to integrate the LivePerson  SDK into my project, but there is a problem and it does not compile:

```

C:\Users\...\AndroidStudioProjects\MyApp\lp_messaging_sdk\build\generated\source\r\debug\com\liveperson\infra\messaging_ui\R.java:1657: error: ';' expected
        public static int lp_message_time_n?ow_with_state = 0x7f070098;
                                           ^
C:\Users\...\AndroidStudioProjects\MyApp\lp_messaging_sdk\build\generated\source\r\debug\com\liveperson\infra\messaging_ui\R.java:1657: error: <identifier> expected
        public static int lp_message_time_n?ow_with_state = 0x7f070098;
                                                         ^
C:\Users\...\AndroidStudioProjects\MyApp\lp_messaging_sdk\build\generated\source\r\debug\com\liveperson\infra\ui\R.java:843: error: ';' expected
        public static int lp_message_time_n?ow_with_state = 0x7f070098;
                                           ^
C:\Users\...\AndroidStudioProjects\MyApp\lp_messaging_sdk\build\generated\source\r\debug\com\liveperson\infra\ui\R.java:843: error: <identifier> expected
        public static int lp_message_time_n?ow_with_state = 0x7f070098;
                                                         ^
4 errors

:lp_messaging_sdk:compileDebugJavaWithJavac FAILED

FAILURE: Build failed with an exception.
```

It looks like there’s an invalid character in the lp_message_time_n?ow_with_state string resource, for one (or more) translations.

This happens on v2.5.1 and v2.7.0

**Comment by dvillacis94** [comment]
Which version of Android are you running?

**Comment by bramley-turner-stride** [comment]
compileSdkVersion: 25,
buildToolsVersion: '25.0.2',
minSdkVersion: 11,
targetSdkVersion: 23,

**Comment by dvillacis94** [comment]
Have you tried updating your minSdkVersion to 19, that is the lowest Android version supported on the In-App Android SDK 2.5 and 2.7

```android 
   defaultConfig {
        minSdkVersion 19
        targetSdkVersion 25
        versionCode 251
        versionName "2.5.1"
    }
```

**Comment by Hauuguu** [comment]
@bramley-turner-stride Are you using Windows or Mac?

**Comment by bramley-turner-stride** [comment]
@Hauuguu Windows

@dvillacis94 no thankyou, that will drop several thousand users. We're simply not using the SDK on devices below 19. I'm using v2.5.0 for now, until the string is fixed.

**Comment by pmolensky** [comment]
I'm building the sample app (2.7.0) for a client and get the same error. And the minSdkVersion is set to 19 in my build.gradle.

```
Warning:string 'lp_message_time_n?ow_with_state' has no default translation.
Warning:string 'lpmessaging_ui_fill_in_form_text_empty' has no default translation.
C:\Users\...\Android-Messaging-SDK-2.7.0\SampleApp-Source\sample_app\lp_messaging_sdk\build\generated\source\r\release\com\liveperson\infra\messaging_ui\R.java
Error:(1657, 44) error: ';' expected
Error:(1657, 58) error: <identifier> expected
C:\Users\...\Android-Messaging-SDK-2.7.0\SampleApp-Source\sample_app\lp_messaging_sdk\build\generated\source\r\release\com\liveperson\infra\ui\R.java
Error:(843, 44) error: ';' expected
Error:(843, 58) error: <identifier> expected
```

**build.gradle**
```
    compileSdkVersion 25
    buildToolsVersion "25.0.2"

    defaultConfig {
        minSdkVersion 19
        targetSdkVersion 25
        versionCode 270
        versionName "2.7.0"
    }
```

**Comment by Hauuguu** [comment]
This issue seems to be fixed in ver 2.8

**Comment by pmolensky** [comment]
v2.8 of the SDK builds without issues, so for me the issue is solved
